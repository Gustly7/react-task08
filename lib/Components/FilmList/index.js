'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _filmApi = require('../../api/film-api');

var filmApi = _interopRequireWildcard(_filmApi);

var _Film = require('../Film/Film');

var _Film2 = _interopRequireDefault(_Film);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FilmList = function (_Component) {
  _inherits(FilmList, _Component);

  function FilmList() {
    _classCallCheck(this, FilmList);

    return _possibleConstructorReturn(this, (FilmList.__proto__ || Object.getPrototypeOf(FilmList)).apply(this, arguments));
  }

  _createClass(FilmList, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var params = new URLSearchParams(this.props.location.search);
      if (params.get('search') != null) {
        filmApi.searchFilms(params.get('search'), params.get('searchBy'));
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { className: 'container filmList' },
        this.props.films.data.map(function (film) {
          return _react2.default.createElement(
            'div',
            { key: film.id, className: 'data-list-item' },
            _react2.default.createElement('div', { className: 'details' }),
            _react2.default.createElement(_Film2.default, { film: film })
          );
        })
      );
    }
  }]);

  return FilmList;
}(_react.Component);

var mapStateToProps = function mapStateToProps(store) {
  return {
    films: store.filmState.films
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps)(FilmList);