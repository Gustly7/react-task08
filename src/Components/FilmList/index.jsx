// @flow
import React, { Component } from 'react'

import { connect } from 'react-redux'

import * as filmApi from '../../api/film-api'

import Film from '../Film/Film'

type film = {
  id: number;
  poster_path: string;
  title: string;
  release_date: string;
  genres: string;
}

type Props = {
  className?: string,
  disabled?: boolean,
  films: {data: any},
  location: { search22: string },
};

class FilmList extends Component<Props> {
  componentDidMount () {
    const params = new URLSearchParams(this.props.location.search)
    if (params.get('search') != null) {
      filmApi.searchFilms(params.get('search'), params.get('searchBy'))
    }
  }

  render ()
  {
    return (<div className='container filmList'>

      {this.props.films.data.map(film => {
        return (
          <div key={film.id} className='data-list-item'>
            <div className='details' />
            <Film film={film} />
          </div>
        )
      })}

    </div>)
  }
}

const mapStateToProps = function (store) {
  return {
    films: store.filmState.films
  }
}

export default connect(mapStateToProps)(FilmList)
