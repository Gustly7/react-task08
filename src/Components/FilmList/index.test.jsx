import React, { Component } from 'react'
import FilmList from './index'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import fetchMock from 'fetch-mock'
import store from '../../store'
import configureStore from 'redux-mock-store'
const mockStore = configureStore()

const middlewares = [thunk]
// const mockStore = configureMockStore(middlewares)

import { Provider } from 'react-redux'

describe('FilmList component test suite', () => {
  const film = {id: 1, name: 'Kill bill', release_date: '01-02-2004', genres: ['Action', 'Adventure'], director: 'Quentin Tarantino', poster_path: 'https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg', rating: 10}
  let store

  afterEach(() => {
    fetchMock.reset()
    fetchMock.restore()
  })

  beforeEach(() => {
    store = mockStore({filmState: {films: {data: [film]}}})
  })

  test('FilmList snapshot renders correctly', () => {
    const tree = renderer.create(
      <FilmList store={store} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('FilmList div should have "container filmList" classes', () => {
    const wrapper = shallow(
      <FilmList store={store} />
    )
    const elem = wrapper.find('div.filmList')
    const elem2 = wrapper.find('div')
    console.log(wrapper)
    console.log(elem)
    console.log(elem2)
    console.log(wrapper.html())
    console.log(wrapper.text())
    expect(wrapper.length).toBe(1)
  })
})
