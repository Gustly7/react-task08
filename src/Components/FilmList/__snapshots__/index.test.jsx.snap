// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList component test suite FilmList snapshot renders correctly 1`] = `
<div
  className="container filmList"
>
  <div
    className="data-list-item"
  >
    <div
      className="details"
    />
    <div
      className="Film"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="Poster"
            >
              <img
                alt={undefined}
                src="https://images-na.ssl-images-amazon.com/images/I/41qSUP7S3XL._AC_UL320_SR228,320_.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="FilmName"
            />
          </div>
          <div
            className="col"
          >
            <div
              className="FilmYear"
            >
              01-0
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="FilmGenre"
            >
              Action
              Adventure
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
